.github/workflows/python_uniffi_ci_cd.yml

```diff
@@ -0,0 +1,275 @@
+
+on:
+  workflow_call:
+    inputs:
+      crate:
+        required: true
+        type: string
+        description: "The name of the crate to build and publish"
+      release:
+        required: false
+        type: boolean
+        default: false
+        description: "Whether to run release steps"
+    secrets:
+      BOT_ID:
+        required: true
+      BOT_SK:
+        required: true
+
+permissions:
+  contents: write # to make commits
+  actions: write # to upload artifacts
+  issues: write # to be able to comment on released issues
+  pull-requests: write # to be able to comment on released pull requests
+
+env:
+  # Python >= 3.8 is needed to support manylinux_x_y tags (unless you want to manually update pip)
+  # Python 3.10 was chosen because it is currently the most popular according to https://mayeut.github.io/manylinux-timeline/
+  PYTHON_VERSION: "3.10"
+
+jobs:
+  get_version:
+    if: inputs.release
+    runs-on: ubuntu-latest
+    outputs: 
+      tag: ${{ steps.semantic_release.outputs.tag }}
+      version: ${{ steps.semantic_release.outputs.version }}
+    defaults:
+      run:
+        working-directory: ./packages/python/${{ inputs.crate }}/
+    steps:
+      - uses: actions/create-github-app-token@v1
+        id: app_token
+        with:
+          app-id: ${{ secrets.BOT_ID }}
+          private-key: ${{ secrets.BOT_SK }}
+      - uses: actions/checkout@v4
+      - uses: oven-sh/setup-bun@v2
+        with:
+          bun-version: latest
+      - run: cd ../../../ && bun install
+      - name: Semantic Release (dry-run)
+        id: semantic_release
+        env:
+          GITHUB_TOKEN: ${{ steps.app_token.outputs.token }}
+        run: "bun semantic-release --dry-run"
+      - name: Update version in pyproject.toml
+        if: steps.semantic_release.outputs.version != ''
+        run: sed -i 's/version = .*/version = "${{ steps.semantic_release.outputs.version }}"/' pyproject.toml
+      - name: Commit version bump
+        if: steps.semantic_release.outputs.version != ''
+        env:
+          GITHUB_TOKEN: ${{ steps.app_token.outputs.token }}
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git add pyproject.toml
+          git commit -m "chore(python/${{ inputs.crate }}): bump version to ${{ steps.semantic_release.outputs.version }}"
+          git push
+
+  build_and_test:
+    needs: get_version
+    if: ${{ !inputs.release || needs.get_version.outputs.version != '' }}
+    defaults:
+      run:
+        shell: bash
+    runs-on: ${{ matrix.target.runner }}
+    strategy:
+      matrix:
+        target:
+          # name: The name of the target passed to cargo build
+          # runner: The GitHub runner to use
+          - name: x86_64-pc-windows-msvc
+            runner: windows-latest
+          # - name: aarch64-pc-windows-msvc
+          #   runner: windows-11-arm
```

Comment:
Just noting that we now have a windows-11-arm runner available, but it doesn't have rustup. After this PR is merged I will experiment and follow-up with a windows-11-arm fix.